zephyr/lib/regions_mm.c

```diff
@@ -1,52 +1,570 @@
 // SPDX-License-Identifier: BSD-3-Clause
 /*
- * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
  *
  * Author: Jakub Dabek <jakub.dabek@intel.com>
  */
 
 #include <sof/lib/regions_mm.h>
 
-struct virtual_memory_heap
-	vm_heaps[CONFIG_MP_MAX_NUM_CPUS + VIRTUAL_REGION_COUNT];
+
+/* list of virtual_memory_heap objects created */
+struct list_item vmh_list;
 
 /**
- * @brief Fill vm_heaps array with information from zephyr.
+ * @brief Initialize new heap
+ *
+ * Initializes new heap based on region that heap is to be assigned to
+ * and config. The heap size overall must be aligned to physical page
+ * size.
+ * @param cfg Configuration structure with block structure for the heap
+ * @param memory_region_attribute A virtual region identifier
+ * @param span Flag deciding if heap can do contiguous allocation.
  *
- * Virtual memory regions calculated in zephyr are translated here
- * to a struct that will keep all information on current allocations
- * and virtual to physical mappings that are related to heaps.
- * System heap is not a part of this information. It only refers to
- * virtual first heaps.
- * Has to be initialized after calculations for regions is done in zephyr.
+ * @retval ptr to a new heap if successful.
+ * @retval NULL on creation failure.
  */
-static int virtual_heaps_init(void)
+struct virtual_memory_heap *vmh_init_heap(const struct heap_config *cfg,
+					  int memory_region_attribute, bool span)
```

Comment:
What do we mean by ```@param memory_region_attribute A virtual region identifier``` is it a pointer ? an index, enum or mask ? The identifer description is not aligning with the attribute in the name.

Reply:
added comment in documentation, the attribute is zephyr provided and identifies a virtual region.